app/src/main/kotlin/com/wire/android/ui/authentication/devices/register/RegisterDeviceViewModel.kt

```diff
@@ -14,63 +14,97 @@ import com.wire.android.navigation.NavigationCommand
+            updateState(state.copy(password = newText, error = RegisterDeviceError.None, continueEnabled = newText.text.isNotEmpty()))
     }
 
     fun onErrorDismiss() {
-        state = state.copy(error = RegisterDeviceError.None)
+        updateErrorState(RegisterDeviceError.None)
+    }
+
+    private suspend fun registerClient(password: String?) {
+        updateState(state.copy(loading = true, continueEnabled = false))
+        when (val registerDeviceResult = registerClientUseCase(
+            RegisterClientUseCase.RegisterClientParam(
+                password = password,
+                capabilities = null,
+            )
+        )) {
+            is RegisterClientResult.Failure.TooManyClients -> navigateToRemoveDevicesScreen()
+            is RegisterClientResult.Success -> {
+                registerPushToken(registerDeviceResult.client.id)
+                navigateToHomeScreen()
+            }
+            is RegisterClientResult.Failure.Generic -> state = state.copy(
+                loading = false,
+                continueEnabled = true,
+                error = RegisterDeviceError.GenericError(registerDeviceResult.genericFailure)
+            )
+            RegisterClientResult.Failure.InvalidCredentials -> state = state.copy(
+                loading = false,
+                continueEnabled = true,
+                error = RegisterDeviceError.InvalidCredentialsError
+            )
+        }
     }
 
     fun onContinue() {
-        state = state.copy(loading = true, continueEnabled = false)
         viewModelScope.launch {
-            if (!validatePasswordUseCase(state.password.text))
-                state = state.copy(loading = false, continueEnabled = true, error = RegisterDeviceError.InvalidCredentialsError)
-            else when (val registerDeviceResult = registerClientUseCase(
-                RegisterClientUseCase.RegisterClientParam(
-                    password = state.password.text,
-                    capabilities = null,
-                ))) {
-                is RegisterClientResult.Failure.TooManyClients -> navigateToRemoveDevicesScreen()
-                is RegisterClientResult.Success -> {
-                    registerPushToken(registerDeviceResult.client.id)
-                    navigateToHomeScreen()}
-                is RegisterClientResult.Failure.Generic -> state = state.copy(
-                        loading = false,
-                        continueEnabled = true,
-                        error = RegisterDeviceError.GenericError(registerDeviceResult.genericFailure)
-                )
-                RegisterClientResult.Failure.InvalidCredentials -> state = state.copy(
-                    loading = false,
-                    continueEnabled = true,
-                    error = RegisterDeviceError.InvalidCredentialsError
-                )
-            }
+            registerClient(state.password.text)
```

Comment:
We are not checking the password is a valid one anymore, even if it is for registering devices?

Reply:
The password validation is only used when the user registers an account or reset the password, validating here is incorrect since users with old password rules should also be allowed to log in